### PR TITLE
Fix #159 - Add support for ip on webserver

### LINKED ISSUE #159: Setup Webserver on RedHat OpenShift
Hello,
I've tried uploading my botkit project on RedShift but it was not working.
I had the following error:
**Application 'appname' failed to start (port 8080 not available)**

The problem is that the express webserver needs an IP on RedShift to get started.

What I've done is I added the IP in the SlackBot.js at line 160 to get the IP from the environment.

```
......
        slack_botkit.webserver = express();

        slack_botkit.webserver.set('ip', process.env.OPENSHIFT_NODEJS_IP || "127.0.0.1");

        slack_botkit.webserver.use(bodyParser.json());
        slack_botkit.webserver.use(bodyParser.urlencoded({ extended: true }));
        slack_botkit.webserver.use(express.static(__dirname + '/public'));

        var server = slack_botkit.webserver.listen(
           slack_botkit.config.port,slack_botkit.webserver.set('ip'), ......

```

I'm not sure if it is the case on other hosting services but this might be worth integrating the **ip** parameter when calling `controller.setupWebserver(port,function(err,webserver)`
Works like a charm otherwise.

More info here: http://stackoverflow.com/questions/31358992/application-appname-failed-to-start-port-8080-not-available-on-open-shift-no

### PR DESCRIPTION
**Pull request by LoneRifle**
This PR is based on - and supersedes - #249, and addresses an issue with that PR, namely, the quoting of the localhost IP as a string. It adds a new supported field - ip - to the config consumed by the Slack variant of Botkit so that it can be used out-of-the-box on hosting providers (like RedHat Openshift) that require processes to listen on specific IP addresses.

This issue fixes #159.

### CODE REVIEW

File: lib/SlackBot.js [hunk 1]
```diff
@@ -217,8 +218,10 @@ function Slackbot(configuration) {
 
         var server = slack_botkit.webserver.listen(
             slack_botkit.config.port,
+            slack_botkit.config.ip,
             function() {
-                slack_botkit.log('** Starting webserver on port ' +
+                slack_botkit.log('** Starting webserver on ' +
+                    ((slack_botkit.config.ip)?slack_botkit.config.ip+':':'port ') +
```
Comment (anonrig) [inline lib/SlackBot.js]: Can you make this line more readable please?

File: lib/SlackBot.js [hunk 2]
```diff
@@ -209,6 +209,7 @@ function Slackbot(configuration) {
             static_dir = slack_botkit.config.webserver.static_dir;
 
         slack_botkit.config.port = port;
+        slack_botkit.config.ip = slack_botkit.config.ip || '127.0.0.1';
```
Comment (anonrig) [inline lib/SlackBot.js]: Can you double check if `slack_botkit.config` exists. If it's not, it will crash eventually.

**Comment by LoneRifle** [comment]
Changes applied in response to comments, please review.

File: lib/SlackBot.js [hunk 3]
```diff
@@ -207,19 +207,24 @@ function Slackbot(configuration) {
...
+        var ip = slack_botkit.config? slack_botkit.config.ip : '127.0.0.1';
 
         slack_botkit.webserver = express();
         slack_botkit.webserver.use(bodyParser.json());
         slack_botkit.webserver.use(bodyParser.urlencoded({ extended: true }));
         slack_botkit.webserver.use(express.static(static_dir));
 
         var server = slack_botkit.webserver.listen(
-            slack_botkit.config.port,
+            port,
```
Comment (anonrig) [inline lib/SlackBot.js]: Why did you change this line to port?

If config exists port is not changed. But, slack_botkit.config.port changes to port. Therefore, there exist a logical error.

Comment (LoneRifle) [inline lib/SlackBot.js]: Can you clarify or rephrase your sentence please? You imply in an earlier diff comment that we cannot assume that slack_botkit.config is defined. Given this, and the fact that we enforce that port is defined earlier in lines 199-201 and will overwrite anything in slack_botkit.config.port, the use of port should not result in any change in behaviour.

**Comment by LoneRifle** [comment]
Change of port argument done as requested, please review.

**Comment by LoneRifle** [comment]
Could I have an update on this PR please?
